### DISCUSSION TITLE
Smart Playlists similar to Symphonium's

### DISCUSSION THREAD

**Discussion by calvinthefreak** [audioling/audioling#20] [Ideas]
Maybe it could be implemented, to get Playlists, that recommend music on your Favourites and similar songs, that you'd like. 

I know, this is maybe a bit more difficult to implement.

**Comment by jeffvli** [comment]
This is also something I want to implement.

![image](https://github.com/user-attachments/assets/158c5f69-9395-4fed-9181-5ad9e49a0d5d)

I've built out a proof of concept implementation of it already, but there's some drawbacks to it due to being limited to the existing APIs.

~~1. **There's no dedicated server API endpoints that can support "smart" querying, which means that your entire library needs to be indexed from the server first before it can be queried.** That means that 1) you need to re-query your entire library every time you want to create a new smart playlist or 2) have the user manually refresh the library cache on demand when they explicitly make changes to it (such as adding or removing albums).~~

~~2. **I initially wanted a way to query without requiring a server playlist be made on every query.** This has some complications due to the issues above with either having to re-index your library every query or running into stale data.~~ Though, this can be resolved by generating (and later deleting) a playlist on the server to retrieve fresh data on every query.

I finished typing up my notes about this feature and then I realized I had a good solution for handling said drawbacks. Going to leave it here for reference.

**Comment by raphmim** [comment]
Also would be cool to pull smart playlists from Navidrome, as it's coming soon: https://github.com/navidrome/navidrome/issues/1417
